projects/plugins/jetpack/extensions/blocks/donations/deprecated/v2/view.js

```diff
@@ -0,0 +1,230 @@
+		const donateButton = this.getDonateButton();
+		enable
+			? donateButton.classList.remove( 'is-disabled' )
+			: donateButton.classList.add( 'is-disabled' );
+	}
+
+	updateUrl() {
+		const donateButton = this.getDonateButton();
+		const url = donateButton.getAttribute( 'href' );
+		if ( this.amount ) {
+			donateButton.setAttribute(
+				'href',
+				addQueryArgs( url, {
+					amount: this.amount,
+					...( this.isCustomAmount && { customAmount: true } ),
+				} )
+			);
+		} else {
+			donateButton.setAttribute( 'href', removeQueryArgs( url, 'amount', 'customAmount' ) );
+		}
+	}
+
+	updateAmountFromCustomAmountInput() {
+		const input = this.block.querySelector( '.donations__custom-amount .donations__amount-value' );
+		const wrapper = this.block.querySelector( '.donations__custom-amount' );
+
+		const amount = input.innerHTML;
+		if ( ! amount ) {
+			this.amount = null;
+			this.toggleDonateButton( false );
+			return;
+		}
+
+		// Validates the amount.
+		const currency = input.dataset.currency;
+		const parsedAmount = parseAmount( amount, currency );
+		if ( parsedAmount && parsedAmount >= minimumTransactionAmountForCurrency( currency ) ) {
+			wrapper.classList.remove( 'has-error' );
+			this.amount = parsedAmount;
+			this.toggleDonateButton( true );
+		} else {
+			wrapper.classList.add( 'has-error' );
+			this.amount = null;
+			this.toggleDonateButton( false );
+		}
+		this.updateUrl();
+	}
+
+	initNavigation() {
+		const navItems = this.block.querySelectorAll( '.donations__nav-item' );
+		const tabContent = this.block.querySelector( '.donations__tab' );
+		const tabContentClasses = {
+			'one-time': 'is-one-time',
+			'1 month': 'is-monthly',
+			'1 year': 'is-annual',
+		};
+
+		const handleClick = event => {
+			// Update selected interval.
+			const prevInterval = this.interval;
+			const newInterval = event.target.dataset.interval;
+			this.interval = newInterval;
+
+			// Toggle nav item.
+			const prevNavItem = this.getNavItem( prevInterval );
+			if ( prevNavItem ) {
+				prevNavItem.classList.remove( 'is-active' );
+			}
+			const newNavItem = this.getNavItem( newInterval );
+			if ( newNavItem ) {
+				newNavItem.classList.add( 'is-active' );
+			}
+
+			// Toggle tab content.
+			tabContent.classList.remove( tabContentClasses[ prevInterval ] );
+			tabContent.classList.add( tabContentClasses[ newInterval ] );
+
+			// Reset chosen amount.
+			this.amount = null;
+			this.isCustomAmount = false;
+			this.resetSelectedAmount();
+			this.updateUrl();
+
+			// Disable donate button.
+			this.toggleDonateButton( false );
+		};
+
+		navItems.forEach( navItem => {
+			navItem.addEventListener( 'click', handleClick );
+			navItem.addEventListener( 'keydown', handleClick );
+		} );
+
+		// Activates the default tab on first execution.
+		const navItem = this.getNavItem( this.interval );
+		if ( navItem ) {
+			navItem.classList.add( 'is-active' );
+		}
+		tabContent.classList.add( tabContentClasses[ this.interval ] );
```

Comment:
This throws an error on the front-end of my Atomic site: 
```
Uncaught TypeError: Cannot read properties of null (reading 'classList')
    at a.initNavigation (??-eJyVjDkOAjEMAD9E1loBQRSIp6DEMci7uRTn+D5pkCihm2JmRlaYYqVYIfv24iiwUc0Gd+Wow4MjgvUJdwGXoqmcptGZxrLJYfwSN/YODCKJsPWkngmbLIHjf4dgpFKxpnxRLVMURX32n+U93FZ9Oq+Xo77qN/I5Utw=:formatted:3881:19)
    at a.init (??-eJyVjDkOAjEMAD9E1loBQRSIp6DEMci7uRTn+D5pkCihm2JmRlaYYqVYIfv24iiwUc0Gd+Wow4MjgvUJdwGXoqmcptGZxrLJYfwSN/YODCKJsPWkngmbLIHjf4dgpFKxpnxRLVMURX32n+U93FZ9Oq+Xo77qN/I5Utw=:formatted:3796:22)
    at ??-eJyVjDkOAjEMAD9E1loBQRSIp6DEMci7uRTn+D5pkCihm2JmRlaYYqVYIfv24iiwUc0Gd+Wow4MjgvUJdwGXoqmcptGZxrLJYfwSN/YODCKJsPWkngmbLIHjf4dgpFKxpnxRLVMURX32n+U93FZ9Oq+Xo77qN/I5Utw=:formatted:3934:29
    at NodeList.forEach (<anonymous>)
    at HTMLDocument.<anonymous> (??-eJyVjDkOAjEMAD9E1loBQRSIp6DEMci7uRTn+D5pkCihm2JmRlaYYqVYIfv24iiwUc0Gd+Wow4MjgvUJdwGXoqmcptGZxrLJYfwSN/YODCKJsPWkngmbLIHjf4dgpFKxpnxRLVMURX32n+U93FZ9Oq+Xo77qN/I5Utw=:formatted:3928:70)
````
Steps to reproduce:
- Create a new donations block.
- Duplicate the block.
- Save the post and go to the front-end.
- The second (duplicated block) it's not loading, it displays a loading bar and in the console you'll get this error.

Reply:
I think I've found the culprit!

In `domReady` we are initializing the whole view class for all blocks with `.wp-block-jetpack-donations` class.

https://github.com/Automattic/jetpack/blob/f9def81e71ae00fa5d8053d9f4eba5794cbeb3c6/projects/plugins/jetpack/extensions/blocks/donations/view.js#L242

But as it happens, each block has _two_ elements with that class.
One is the block's wrapper (which is what we want to target), using `Blocks::classes()` to autogenerate the class:

https://github.com/Automattic/jetpack/blob/f9def81e71ae00fa5d8053d9f4eba5794cbeb3c6/projects/plugins/jetpack/extensions/blocks/donations/donations.php#L142-L158

The other is the saved content of the block, with the block's class injected automatically by Gutenberg:

https://github.com/Automattic/jetpack/blob/f9def81e71ae00fa5d8053d9f4eba5794cbeb3c6/projects/plugins/jetpack/extensions/blocks/donations/save.js#L7-L15

When the `domReady` callback cycles through all the selected elements, it manages to init the class for the first one (which is, correctly, the first block's wrapper), then it breaks at the second element (which is the content inside the first block's wrapper).

This is easily testable by trying this in the `domReady`:

```js
const blocks = document.querySelectorAll( '.wp-block-jetpack-donations:not(.donations__content)' );
```

The culprit for the block's class injected into the saved content is Gutenberg's [`supports.className`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#classname) property.

In this case, since we already add the class to the wrapper in PHP we might want to remove support for the `className`:
```js
{
  supports: {
    className: false,
  },
}
```

Reply:
In this [PR Border Color for Donations Block](https://github.com/Automattic/jetpack/pull/22946), the structure of the block has been changed in order to provide the requested functionality. As a side effect of this change, this issue no longer happens. I'll ignore this problem for now.